nvmeadm/src/nvmf_discovery.rs

```diff
@@ -426,91 +440,129 @@ impl DiscoveryBuilder {
                 return Err("invalid transport specified".into());
             }
         }
-
         if let Some(traddr) = &self.traddr {
-            if let Err(ip) = IpAddr::from_str(&traddr) {
-                return Err(format!("Invalid IP address: {}", ip.to_string()));
-            }
+            is_valid_ip(&traddr)?;
         }
 
         Ok(())
     }
 }
 
-impl DiscoveryLogEntry {
-    pub fn build_connect_args(&mut self) -> Result<String, NvmeError> {
-        let mut connect_args = String::new();
-        let host_id = HOST_ID.as_str();
-
-        connect_args.push_str(&format!("nqn={},", self.subnqn));
-        connect_args.push_str(&format!(
-            "hostnqn={},",
-            format!("nqn.2019-05.io.openebs.mayastor:{}", host_id)
-        ));
-        connect_args.push_str(&format!("hostid={},", host_id));
+#[derive(Default, Debug, Builder)]
+#[builder(setter(into))]
+#[builder(build_fn(validate = "Self::validate"))]
+pub struct ConnectArgs {
+    /// IP address of the target
+    traddr: String,
+    /// Port number of the target
+    trsvcid: String,
```

Comment:
NvmeTarget uses u16 for the ports, would it make sense to do the same here?

Reply:
we need to keep this structure transport independent because it can be used for rdma transports as well. Not from our CSI code but in target discovery code that is in nvmeadm lib.